packages/opentelemetry-tracing/src/platform/browser/export/BatchSpanProcessor.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BatchSpanProcessorBase } from '../../../export/BatchSpanProcessorBase';
+
+export class BatchSpanProcessor extends BatchSpanProcessorBase {
+  private _onVisibilityChange(): void {
+    if (document.visibilityState === 'hidden') {
+      void this.forceFlush();
+    }
+  }
+
+  onInit(): void {
```

Comment:
Two things.

1.  In web worker the document in undefined, so I would add here checking for document existence and if that doesn't exist then do nothing so this way it remain functioning in web worker (same for shutdown).


2. Because we will change the current behaviour someone might want to be able to disable (for various reasons) I think it make sense to enable this by default, but on the same time I would add config param which will allow user to disable this, so that no1 will be "forced" to use new behaviour.

Reply:
Good catch. I called the config option `flushOnDocumentBecomesHidden`.

Reply:
by default it will be enabled so in fact you will be disabling the default behaviour by setting the param to be true. This way I would put the name disable into this name, maybe `disableAutoFlashOnDocumentHide` ?

Reply:
Good idea. Changed to `disableAutoFlushOnDocumentHide`.